frontend/src/pages/auth/signup/SignUp.tsx

```diff
@@ -1,12 +1,153 @@
-import React from 'react';
+import React, {useState} from 'react';
+import {Box, Button, Checkbox, Link, TextField, Typography} from '@mui/material';
+import {Paths} from "../../../common/constansts/paths";
 import {Titles} from "../../../common/constansts/titles";
+import Logo from "../../../img/Logo.png";
+import "../../../common/styles/button.css";
+import "./styles.css";
+import {useNavigate} from "react-router-dom";
+import {UserCredentials} from "../../../types/userCredentials";
+import {Buttons} from "../../../common/constansts/buttons";
+import {Messages} from "../../../common/constansts/messages";
+import {Roles} from "../../../common/constansts/roles";
+import {UserType} from "../../../types/userType";
 
 const SignUp: React.FC = () => {
+    const [email, setEmail] = useState('');
+    const [password, setPassword] = useState('');
+    const [name, setName] = useState('');
+    const navigate = useNavigate();
+    const [customerChecked, setCustomerChecked] = useState(false);
+    const [merchantChecked, setMerchantChecked] = useState(false);
+    const [agreeChecked, setAgreeChecked] = useState(false);
+
+    const handleCustomerCheckboxChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setCustomerChecked(event.target.checked);
+        if (event.target.checked) {
+            setMerchantChecked(false);
+        }
+    };
+
+    const handleMerchantCheckboxChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setMerchantChecked(event.target.checked);
+        if (event.target.checked) {
+            setCustomerChecked(false);
+        }
+    };
+
+    const handleAgreeCheckboxChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setAgreeChecked(event.target.checked);
+    };
+
+    const handleSignUp = () => {
+        const isSignUpEnabled = email && name && password && (customerChecked || merchantChecked) && agreeChecked;
```

Comment:
Create separate method which validates all cases.

Btw, your email and password are not validated correctly. Set some simple rules for password, e.g. at least 4 symbols. If you already have some rules on the backend, sync it. Same for email.

If you think, why do we need to validate it on the client if we do the same on the server the answer is simple, we don't want to bother/load our servers with unnecessary calls.

But if we validate these fields on the client why do we need to validate it on the server? We can never rely on what client sends us, because user can be hacked. Or we expose our api for 3rd party clients and they can do poor validation.